services/workflows-service/src/common/filters/AllExceptions.filter.ts

```diff
@@ -77,6 +78,15 @@ export class AllExceptionsFilter extends BaseExceptionFilter {
       error: inspect(errorRes),
       message: error.message,
       responseTime: Date.now() - request.startTime,
+      url: request.url,
+      method: request.method,
+      headers: removeSensitiveHeaders(request.headers),
+      query: request.query,
+      body: request.body,
```

Comment:
super risky should check that the http content is json

Reply:
you can use `getReqMetadataObj` and also add the suggested body manipulation for json only 

```
export const getReqMetadataObj = (req: Request<unknown>) => {
  const cleanHeaders = removeSensitiveHeaders(req.headers);
  const isJsonPayload = req.headers['content-type']?.includes('application/json');
  
  // Check if the body is a valid JSON (object or array, but not null)
  const isValidBody = isJsonPayload && req.body !== null && (typeof req.body === 'object');

  return {
    startTime: req.startTime ? new Date(req.startTime)?.toISOString() : new Date().toISOString(),
    'X-Request-ID': req.id,
    query: req.query,
    params: req.params,
    url: req.originalUrl,
    method: req.method,
    headers: cleanHeaders,
    body: isValidBody ? req.body : undefined, // Log only if valid JSON (object or array)
  };
};
```

Reply:
good point, done